crates/parcel/src/request_tracker/request_tracker.rs

```diff
@@ -14,19 +17,36 @@ use super::RunRequestError;
 
 pub struct RequestTracker<T> {
   graph: RequestGraph<T>,
+  reporters: Vec<Box<dyn ReporterPlugin>>,
   request_index: HashMap<u64, NodeIndex>,
 }
 
+impl<T: Clone> Default for RequestTracker<T> {
+  fn default() -> Self {
+    RequestTracker::new(Vec::new())
+  }
+}
+
 impl<T: Clone> RequestTracker<T> {
-  pub fn new() -> Self {
+  pub fn new(reporters: Vec<Box<dyn ReporterPlugin>>) -> Self {
     let mut graph = StableDiGraph::<RequestNode<T>, RequestEdgeType>::new();
     graph.add_node(RequestNode::Root);
     RequestTracker {
       graph,
+      reporters,
       request_index: HashMap::new(),
     }
   }
 
+  pub fn report(&self, event: ReporterEvent) {
+    let event = Arc::new(event);
+
+    for reporter in self.reporters.iter() {
+      // TODO Decide how to handle errors
+      let _ = reporter.report(Arc::clone(&event));
```

Comment:
We should just pass the event by reference instead of using `Arc`

`Arc` is for multi-threaded scenarios ; for single-threaded reference counting we can use `Rc`

But for this use-case we should be passing the events by reference into the reporters.

Reply:
I was trying to factor in multi-threading use-case from the get go, but realised I can just do that anyway later with a borrow to reference so I will fix it shortly.